lib/gringotts/money.ex

```diff
@@ -0,0 +1,39 @@
+defprotocol Gringotts.Money do
+  @moduledoc """
+  Money protocol used by the Gringotts API.
+
+  The `amount` argument required for some of Gringotts' API methods must
+  implement this protocol.
+
+  If your application is already using a supported Money library, just pass in
+  the Money struct and things will work out of the box.
+
+  Otherwise, just wrap your `amount` with the `currency` together in a `Map` like so,
+      money = %{amount: Decimal.new(2017.18), currency: "USD"}
+
+  and the API will accept it (as long as the currency is valid [ISO 4217 currency
+  code](https://www.iso.org/iso-4217-currency-codes.html)).
+  """
+  @fallback_to_any true
+  @type t :: Gringotts.Money.t
+  
+  @spec currency(t) :: String.t
+  @doc """
+  Returns the ISO 4217 compliant currency code associated with this sum of money.
+
+  This must be an UPCASE `string`
+  """
+  def currency(money)
+
+  @spec amount(t) :: Decimal.t
+  @doc """
+  Returns a Decimal representing the "worth" of this sum of money in the
+  associated `currency`.
+  """
+  def amount(money)
+end
+
+defimpl Gringotts.Money, for: Any do
+  def currency(money), do: Map.get(money, :currency)
```

Comment:
May I suggest calling `to_string` on the map result which would then normalise from libs (like ex_money) that use an atom for currency code?

```
def currency(money), do: Map.get(money, :currency) |> to_string
```

Reply:
I think this is covered in 1076a9f638b13dab2e8abbb15a241ce636e8f4e4 _(in this PR)_
Both `money` and `ex_money` use upcase `atom`s, so the implementation for `Money` (not `Any`) will fire and give us a `string`. `monetized` uses upcase strings, so there's no need to do an `upcase`.